euphoria-hadoop/src/main/java/cz/seznam/euphoria/hadoop/input/HadoopSplit.java

```diff
@@ -83,33 +83,49 @@ public HadoopSplit(HadoopSource<K, V> source, InputSplit inputSplit) {
       reader.initialize(inputSplit, ctx);
       return new HadoopReader<>(
           reader, source.getKeyClass(), source.getValueClass(), source.getConf());
-    } catch (Exception e) {
+    } catch (InterruptedException e) {
       Thread.currentThread().interrupt();
       throw new IOException("Interrupted while opening reader.");
```

Comment:
I would suggest to add also the root exception to stack trace to notice user it was originally Interrupted Exception so it's not so standard IOE -> throw new IOException("Interrupted while opening reader.", e );

Reply:
:+1:

Reply:
@cervebar Because the branch is already closed, could you please open a separate PR for this? Thanks a lot!